### PR TITLE
tests for form components

### PR DESCRIPTION
**Pull request by nemesis09**
## Fixes 
https://issues.redhat.com/browse/HAC-397

## Description
This PR adds tests for form components added from console.

## Type of change
- [x] Refactoring (no functional changes, no api changes)
- [x] Other (please describe): Tech Debt (add tests: no UI changes involved)

## How to test or reproduce?
`npm test` to run all tests.

## Browser conformance: 
- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+const mockUseTranslation = useTranslation as jest.Mock;
+
+beforeEach(() => {
+  mockUseTranslation.mockImplementation(() => ({ t: (x) => x }));
+});
+
+afterEach(() => {
+  jest.resetAllMocks();
+  cleanup();
```
Comment (invincibleJai) [inline src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx:42]: do we need to explicitly call `cleanup`?

https://testing-library.com/docs/react-testing-library/api/#cleanup

File: src/shared/components/form-components/__tests__/ActionGroupWithIcons.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+jest.mock('react-i18next', () => ({
+  useTranslation: jest.fn(),
+}));
+
+const onSubmit = jest.fn();
+const onClose = jest.fn();
+const mockUseTranslation = useTranslation as jest.Mock;
+
+configure({ testIdAttribute: 'data-test' });
```
[the remark this answers is not in the document]
Comment (nemesis09) [reply src/shared/components/form-components/__tests__/ActionGroupWithIcons.spec.tsx:15]: I think we could that as a part of the ticket to replace enzyme.

File: src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx (hunk 1 above)
Comment (nemesis09) [reply src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx]: we wouldn't need to explicitly call cleanup. removed it from after each block.
Thanks for catching it.

File: src/shared/components/formik-fields/EditableLabelField.tsx [hunk 3]
```diff
@@ -53,7 +53,7 @@ const EditableLabelField: React.FC<EditableLabelFieldProps> = ({
         </Flex>
       ) : (
         <div className="hacDev-editable-label-field__label">
-          {value} {!label && editIcon}
+          {value} {editIcon}
```
Comment (rottencandy) [inline src/shared/components/formik-fields/EditableLabelField.tsx:56]: Target port doesn't seem to be showing edit icon anymore:
(`editIcon` is also used in line 36)

File: src/shared/components/formik-fields/__tests__/DropdownField.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        <DropdownField name={fieldName} label="label" helpText="helpText" items={items} />
+      </Wrapper>,
+    );
+    expect(screen.queryByRole('menu')).toBeFalsy();
+    fireEvent.click(screen.getByText(initialValues.dropdownValue));
+    await waitFor(() => {
+      expect(screen.getByRole('menu')).toBeInTheDocument();
+      const menuItems = screen.getAllByRole('menuitem');
+      const isItemExist = (item) => menuItems.some((mi) => mi.nodeValue === item.value);
+      expect(items.every(isItemExist));
```
Comment (rottencandy) [inline src/shared/components/formik-fields/__tests__/DropdownField.spec.tsx:62]: Are we checking that `nodeValue` exists for all items in the menu? This expect call seems incomplete.
```suggestion
      expect(items.every(isItemExist)).toBeTruthy();
```

File: src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    render(
+      <Wrapper initialValues={initialValues} onSubmit={jest.fn()}>
+        <EditableLabelField name={fieldName} />
+      </Wrapper>,
+    );
+    expect(screen.getByText(initialLabel)).toBeInTheDocument();
+    expect(screen.getByTestId('pencil-icon')).toBeInTheDocument();
+  });
+
+  it('renders field with value but not edit icon if label is provided', () => {
```
Comment (rottencandy) [inline src/shared/components/formik-fields/__tests__/EditableLabelField.spec.tsx:57]: If label is provided, the edit icon should instead be shown next to the label.

File: src/shared/components/form-components/__tests__/FormFooter.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+  });
+
+  it('should not show submit button if hideSubmit is true', () => {
+    render(<FormFooter {...props} hideSubmit />);
+    expect(screen.queryByTestId('submit-button')).toBeFalsy();
+  });
+
+  it('should render submit button with type button if hideSubmit is false and handleSubmit is defined', () => {
+    render(<FormFooter {...props} />);
+    expect(screen.getByTestId('submit-button').getAttribute('type')).toEqual('button');
```
Comment (rottencandy) [inline src/shared/components/form-components/__tests__/FormFooter.spec.tsx:149]: Nit:
```suggestion
    expect(screen.getByTestId('submit-button')).toHaveAttribute('type', 'button');
```

File: src/shared/components/form-components/__tests__/ActionGroupWithIcons.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    });
+  });
+
+  it('fires onClose callback on close button click', async () => {
+    render(<ActionGroupWithIcons onSubmit={onSubmit} onClose={onClose} />);
+    fireEvent.click(screen.getByTestId('close-icon'));
+    await waitFor(() => {
+      expect(onClose).toHaveBeenCalled();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (nemesis09) [reply src/shared/components/form-components/__tests__/ActionGroupWithIcons.spec.tsx:58]: Indeed we do not need to wait here.
Thanks for catching this!

File: src/shared/components/formik-fields/EditableLabelField.tsx (hunk 3 above)
Comment (nemesis09) [reply src/shared/components/formik-fields/EditableLabelField.tsx:56]: fixed

**Review by rottencandy — APPROVED** [review]
Thanks!
/lgtm

**Pull request merged**